deploy/common/deploy.sh

```diff
@@ -42,13 +44,43 @@ echo "Image flavor for roxctl set to $ROXCTL_ROX_IMAGE_FLAVOR"
 
 popd
 
-function curl_central() {
-	cmd=(curl --retry 10 --retry-delay 10 --retry-connrefused --silent --show-error --insecure)
-	local admin_user="${ROX_ADMIN_USER:-admin}"
-	if [[ -n "${ROX_ADMIN_PASSWORD:-}" ]]; then
-		cmd+=(-u "${admin_user}:${ROX_ADMIN_PASSWORD}")
-	fi
-	"${cmd[@]}" "$@"
+# curl_central_once
+# Runs curl with --silent --show-error --insecure.
+# Obeys $ROX_ADMIN_USER and $ROX_ADMIN_PASSWORD.
+# arguments:
+#   - anything else `curl` will accept, including the URL
+function curl_central_once() {
+    cmd=(curl --silent --show-error --insecure)
+    local admin_user="${ROX_ADMIN_USER:-admin}"
+    if [[ -n "${ROX_ADMIN_PASSWORD:-}" ]]; then
+        cmd+=(-u "${admin_user}:${ROX_ADMIN_PASSWORD}")
+    fi
+    "${cmd[@]}" "$@"
+}
+
+
+# curl_central_retry
+# Runs curl like curl_central_once, but with retries and --fail.
+# Captures stdout and emits it only once (for the last invocation of curl).
+# arguments:
+#   - anything else `curl` will accept, including the URL
+function curl_central_retry() {
+    local tmp_out
+    tmp_out="$(mktemp)"
+    local delay_sec=10
+    # On top of --retry we do our own retry loop, to make sure no corner cases slip through.
+    # https://github.com/curl/curl/issues/6712#issuecomment-796534491
+    for _ in $(seq 3); do
+        if "${cmd[@]}" --fail --retry 3 --retry-delay "${delay_sec}" --retry-connrefused "$@" > "${tmp_out}"; then
```

Comment:
Didn't you intend to call `curl_central_once` here instead of `${cmd[@]}`?

Reply:
Oh, good catch! What a coincidence that is still works!